### DISCUSSION TITLE
Collapsible state of child tree items persists after refreshing entry

### DISCUSSION THREAD

**Discussion by david-tejada** [microsoft/vscode-discussions#611] [Extension Development QnA] [answered]
I don't know if I'm not understanding completely how this work but I would expect that after calling `this._onDidChangeTreeData.fire(entry)` the child entries are collapse since when calling `getChildren` all the folder entries are created as collapsed. In the debugger I can actually see that the entries are created as collapsed so I don't understand why they remain expanded.

Here is a [demo repository](https://github.com/david-tejada/vscode-refresh-treeitem) to showcase what I mean.

![demo](https://github.com/microsoft/vscode-discussions/assets/3219671/a5aa1578-a7f2-446e-9536-847a0d9f5cd3)

**Comment by isidorn** [comment]
@alexr00 might be able to help here

**Reply by david-tejada** [reply]
Yeah, she did! I opened an issue and she cleared it up.

**Reply by isidorn** [reply]
Can you please link to that issue so that future users that come upon this discussion can follow the link. Thank you

**Comment by david-tejada** [comment, marked as answer]
Answered here: https://github.com/microsoft/vscode/issues/183360
